src/pocketmine/Player.php

```diff
@@ -3590,6 +3591,25 @@ public function sendTranslation($message, array $parameters = []){
 		$this->dataPacket($pk);
 	}
 
+	public function sendWeather(){
+		$pk1 = new LevelEventPacket();
+		$pk2 = new LevelEventPacket();
+		if($this->level->getWeather() == Level::WEATHER_RAIN_THUNDER){
+			$pk1->evid = LevelEventPacket::EVENT_START_RAIN;
+			$pk2->evid = LevelEventPacket::EVENT_START_THUNDER;
+		}elseif($this->level->getWeather() == Level::WEATHER_RAIN){
+			$pk1->evid = LevelEventPacket::EVENT_START_RAIN;
+			$pk2->evid = LevelEventPacket::EVENT_STOP_THUNDER;
+		} else {
+			$pk1->evid = LevelEventPacket::EVENT_STOP_RAIN;
+			$pk2->evid = LevelEventPacket::EVENT_STOP_THUNDER;
+		}
+		$pk1->data = mt_rand(90000, 110000); //If we're clearing the weather, it doesn't matter what strength values we set
+		$pk2->data = mt_rand(30000, 40000);
+		$this->dataPacket($pk1);
+		$this->dataPacket($pk2);
+	}
```

Comment:
I didn't intend to copy it explicitly. I was looking for a range of values and Genesis just happened to have some.